front/src/components/BoxReconciliationOverlay/components/BoxReconciliationAccordion.tsx

```diff
@@ -55,8 +56,14 @@ export function BoxReconcilationAccordion({
   onBoxUndelivered,
   onBoxDelivered,
 }: IBoxReconcilationAccordionProps) {
+  const [reconciliationMatchProductCache, setReconciliationMatchProductCache] = useAtom(
+    reconciliationMatchProductAtom,
+  );
+  const hasProductCache = useAtomValue(hasReconciliationMatchProductAtomCached);
```

Comment:
@fhenrich33 I would replace the new read-only atom with what you do one level below:

```
const isProductIdMatchedInCache = !!cachedReconciliationMatchProduct[matchingProductSourceId];
```